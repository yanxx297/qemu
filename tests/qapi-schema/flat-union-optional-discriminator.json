# FIXME: we should require the discriminator to be non-optional
{ 'enum': 'Enum', 'data': [ 'one', 'two' ] }
{ 'type': 'Base',
  'data': { '*switch': 'Enum' } }
{ 'type': 'Branch', 'data': { 'name': 'str' } }
{ 'union': 'MyUnion',
  'base': 'Base',
  'discriminator': '*switch',
  'data': { 'one': 'Branch',
            'two': 'Branch' } }
