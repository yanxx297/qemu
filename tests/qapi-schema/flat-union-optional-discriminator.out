[OrderedDict([('enum', 'Enum'), ('data', ['one', 'two'])]),
 OrderedDict([('type', 'Base'), ('data', OrderedDict([('*switch', 'Enum')]))]),
 OrderedDict([('type', 'Branch'), ('data', OrderedDict([('name', 'str')]))]),
 OrderedDict([('union', 'MyUnion'), ('base', 'Base'), ('discriminator', '*switch'), ('data', OrderedDict([('one', 'Branch'), ('two', 'Branch')]))])]
[{'enum_name': 'Enum', 'enum_values': ['one', 'two']}]
[OrderedDict([('type', 'Base'), ('data', OrderedDict([('*switch', 'Enum')]))]),
 OrderedDict([('type', 'Branch'), ('data', OrderedDict([('name', 'str')]))])]
